File: mobile/android/app/src/main/java/com/navtin/engine/GameState.kt
package com.navtin.engine

import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.getValue
import androidx.compose.runtime.setValue

enum class GamePhase {
    PLACING, MOVING, FLYING
}

enum class PlayerTurn {
    PLAYER_1, PLAYER_2
}

enum class GameMode {
    VS_AI, LOCAL, ONLINE
}

enum class PlayerSide {
    WHITE, BLACK
}

class GameState {
    var mode by mutableStateOf(GameMode.LOCAL)
    var aiDifficulty by mutableStateOf(AIDifficulty.MEDIUM)
    var phase by mutableStateOf(GamePhase.PLACING)
    var turn by mutableStateOf(PlayerTurn.PLAYER_1)
    var piecePositions = mutableMapOf<Int, PieceType>()
    
    // Online State
    var roomId by mutableStateOf<String?>(null)
    var playerSide by mutableStateOf<PlayerSide?>(null)
    var userId by mutableStateOf(java.util.UUID.randomUUID().toString())
    var opponentName by mutableStateOf("Opponent")
    
    val isMyTurn: Boolean get() {
        if (mode != GameMode.ONLINE) return true
        if (playerSide == PlayerSide.WHITE && turn == PlayerTurn.PLAYER_1) return true
        if (playerSide == PlayerSide.BLACK && turn == PlayerTurn.PLAYER_2) return true
        return false
    }
    
    var player1PiecesLeftToPlace by mutableStateOf(9)
    var player2PiecesLeftToPlace by mutableStateOf(9)
    
    var player1PiecesOnBoard by mutableStateOf(0)
    var player2PiecesOnBoard by mutableStateOf(0)
    
    var isRemovingPiece by mutableStateOf(false)
    
    // UI state
    var selectedNodeIndex by mutableStateOf<Int?>(null)
    var message by mutableStateOf("Player 1: Place a piece")
    
    val currentPlayer: PlayerTurn get() = turn
    val currentPieceType: PieceType get() = if (turn == PlayerTurn.PLAYER_1) PieceType.PLAYER_1 else PieceType.PLAYER_2
}

File: mobile/ios/NavTin/Engine/GameState.swift
import Foundation

enum GamePhase {
    case placing
    case moving
    case flying
}

enum PlayerTurn {
    case player1
    case player2
}

enum GameMode {
    case vsAI
    case local
    case online
}

enum PlayerSide: String {
    case white
    case black
}

class GameState: ObservableObject {
    @Published var mode: GameMode = .local
    @Published var aiDifficulty: AIDifficulty = .medium
    @Published var phase: GamePhase = .placing
    @Published var turn: PlayerTurn = .player1
    @Published var piecePositions: [Int: PieceType] = [:]
    
    // Online State
    @Published var roomId: String? = nil
    @Published var playerSide: PlayerSide? = nil
    @Published var userId: String = UUID().uuidString
    @Published var opponentName: String = "Opponent"
    @Published var isMyTurn: Bool {
        guard mode == .online else { return true }
        if playerSide == .white && turn == .player1 { return true }
        if playerSide == .black && turn == .player2 { return true }
        return false
    }
    
    @Published var player1PiecesLeftToPlace = 9
    @Published var player2PiecesLeftToPlace = 9
    
    @Published var player1PiecesOnBoard = 0
    @Published var player2PiecesOnBoard = 0
    
    @Published var isRemovingPiece = false
    
    // UI state
    @Published var selectedNodeIndex: Int? = nil
    @Published var message: String = "Player 1: Place a piece"
    
    var currentPlayer: PlayerTurn { turn }
    var currentPieceType: PieceType { turn == .player1 ? .player1 : .player2 }
}
